### PR TITLE
Makefile and example echo server

### PR DESCRIPTION
**Pull request by gotitmemorize**
Added new files that create a TCP asynchronous echo server

Created a makefile to compile new code and parser code

Note that I did not create object files to avoid "littering" the directory. But if we want to create object files, there are a number of ways this can be solved.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+CC=c++
+
+#link -lboost_system library
+CFLAGS = -std=c++0x -g -Wall
```
Comment (alfredlucero) [inline Makefile]: Can call it CCXFLAGS and change the -std=c++11 -g -Wall -Werror

File: Makefile [hunk 2]
```diff
@@ -0,0 +1,22 @@
+CC=c++
+
+#link -lboost_system library
```
Comment (alfredlucero) [inline Makefile]: For style of comments we can do something like this: # Comment goes here.

File: Makefile [hunk 3]
```diff
@@ -0,0 +1,26 @@
+CC=c++
+
+#compile flags
+CFLAGS = -std=c++0x -g -Wall
```
Comment (Wizzard033) [inline Makefile]: -std=c++11

File: Makefile [hunk 4]
```diff
@@ -0,0 +1,26 @@
+CC=c++
```
Comment (Wizzard033) [inline Makefile]: Unnecessary because automatically set

**Comment by Wizzard033** [comment]
Add comments everywhere and put the coding style in there.

Coding style:
(1) Use 4 spaces to indent (no tab characters)
(2) all_lower_case_variable_names_class_names_function_names(int and_args);
(3) Put an underscore after variables that are a part of a class or struct i.e. variable_
(4) Use attached block separators i.e. int main() { has the curly brace on the same line
(5) Two empty lines between things in the outermost scope that would normally have one empty line
(6) Comments above every function/method in non-header files
(7) Comments after every member variable
(8) Comments start with a space and a capital letter

```
// The entry point of the program
int main() {
    blah();
}


// This is a class
class a_class() {
    // Returns the int passed to it
    int return_an_int(int the_int_to_return) {
        return the_int_to_return;
    }

    int value; // A useless value that does nothing
};
```

File: Makefile (hunk 4 above)
Comment (gotitmemorize) [inline Makefile]: good call. Variable used is actually CXX (which is g++)

**Comment by gotitmemorize** [comment]
Cleaned up the makefile and followed suggestions.

Also added comments to explain (as best as I can) some of the logic being used with the boost library

**Pull request merged**